Library/DiscUtils.Vhd/DynamicStream.cs

```diff
@@ -336,7 +330,7 @@ public override int Read(byte[] buffer, int offset, int count)
                 {
                     int toRead = Math.Min(maxToRead - numRead, (int)(_dynamicHeader.BlockSize - offsetInBlock));
                     _parentStream.Position = _position;
-                    StreamUtilities.ReadFully(_parentStream, buffer, offset + numRead, toRead);
+                    StreamUtilities.ReadExact(_parentStream, buffer, offset + numRead, toRead);
```

Comment:
Assumption is made that all bytes were read, so correct behavior is to throw on a short read.